clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -507,6 +507,7 @@
     private final Metadata metadata;
     private final long retryBackoffMs;
     private final long requestTimeoutMs;
+    private final boolean autoCommitEnabled;
```

Comment:
This field is still useful for unit tests (in `KafkaConsumerTest`) that need consumers with auto-commit enabled or disabled. Currently, we seem to be passing the flag to the `KafkaConsumer` constructor but not using it. Unless you think we should handle those scenarios different (e.g. using `ConsumerConfig`)?